service/src/impls.rs

```diff
@@ -220,7 +222,7 @@ pub fn put_value_impl(key: String, value: String, current_timestamp_sec: u64, re
 
     check_key_existence(&connection, key.clone(), current_timestamp_sec.clone())?;
 
-    let values: Vec<Record> = get_values_helper(&connection, key.clone())?.into_iter().filter(|item| item.peer_id == item.set_by).collect();
+    let values: Vec<Record> = get_values_helper(&connection, key.clone())?.into_iter().filter(|item| item.peer_id != call_parameters.host_id).collect();
```

Comment:
Add a comment on what this filtering means, plz.

Reply:
Also, I'm not sure why you're loading all values. It seems that you're only using `values.len()` and `values.iter().last()`. Isn't it possible to do with SQL?